# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../bluez-hcidump/0003-Add-check-for-ACL_START_NO_FLUSH.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 3b1e22fbe09ec0e3cbdbf535b652148a19369c1c Mon Sep 17 00:00:00 2001
From: Daniel Orstadius <daniel.orstadius@nokia.com>
Date: Thu, 10 Feb 2011 12:34:01 +0200
Subject: [PATCH 3/6] Add check for ACL_START_NO_FLUSH

In addition to ACL_START, hcidump should check for the flag
ACL_START_NO_FLUSH to indicate the start of a frame.

Using '==' instead of '&' for the comparison since
ACL_START_NO_FLUSH is defined to zero.

The flag was introduced in BlueZ commit
2430512c983cad8c20252f1df8f297399993ca3d
---
 parser/l2cap.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/parser/l2cap.c b/parser/l2cap.c
index 5c5371f..673aed6 100644
--- a/parser/l2cap.c
+++ b/parser/l2cap.c
@@ -934,7 +934,7 @@ void l2cap_dump(int level, struct frame *frm)
 	l2cap_hdr *hdr;
 	uint16_t dlen;
 
-	if (frm->flags & ACL_START) {
+	if ((frm->flags & ACL_START) || frm->flags == ACL_START_NO_FLUSH) {
 		hdr  = frm->ptr;
 		dlen = btohs(hdr->len);
 
-- 
1.7.2.3

